src/TextInput/TextInput.tsx

```diff
@@ -24,11 +25,11 @@ export type TextInputNonPassthroughProps = {
   /**
    * A visual that renders inside the input before the typing area
    */
-  leadingVisual?: string | React.ComponentType<React.PropsWithChildren<{className?: string}>>
+  leadingVisual?: React.ElementType | React.ReactNode
```

Comment:
we don't pass props internally, so we don't need to actually care what the props the element can take are

Reply:
ooo interesting.  I didn't realize ReactNode included that, and i'm suprised it does (since ElementType isn't renderable as children directly).  I think i'd prefer this since it's a bit more explicit that this could be a Component, but could also be renderable content?